agent-bootstrap/src/main/java/io/opentelemetry/auto/bootstrap/instrumentation/decorator/HttpServerTracer.java

```diff
@@ -218,23 +234,24 @@ private void setStatus(Span span, int status) {
     span.setStatus(HttpStatusConverter.statusFromHttpStatus(status));
   }
 
-  protected abstract Integer peerPort(REQUEST request);
+  protected abstract Integer peerPort(CONNECTION connection);
 
-  protected abstract String peerHostIP(REQUEST request);
+  protected abstract String peerHostIP(CONNECTION connection);
 
   protected abstract HttpTextFormat.Getter<REQUEST> getGetter();
 
   protected abstract URI url(REQUEST request) throws URISyntaxException;
 
   protected abstract String method(REQUEST request);
 
-  /** Stores given context in the given request in implementation specific way. */
-  protected abstract void attachContextToRequest(Context context, REQUEST request);
+  /** Stores given context in the given storage in implementation specific way. */
+  protected abstract void attachServerSpanContext(Context context, STORAGE storage);
```

Comment:
i found "server span context" a bit confusing. i got used to it by the end of reviewing the PR, but maybe just "server context"?

and this method, being defined in `HttpServerTracer` could be just

```suggestion
  protected abstract void attachContext(Context context, STORAGE storage);
```

Reply:
I can rename `attach` method because it is protected, but `getServerSpanContext` is public, meant to be used by instrumentations. And I think there the longer, more explicit name, is better.

Reply:
what do u think of  `attachServerContext()` and `getServerContext()`?